src/access/AccessRestriction.cpp

```diff
@@ -0,0 +1,280 @@
+    }
+
+    newIndex = mFabricEntries[fabricIndex].size() - 1;
+
+    for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+    {
+        listener->OnEntryChanged(fabricIndex, newIndex, localEntry, EntryListener::ChangeType::kAdded);
+    }
+
+    if (index != nullptr)
+    {
+        *index = newIndex;
+    }
+
+    ChipLogProgress(DataManagement, "AccessRestriction: update entry f=%u i=%lu", fabricIndex, newIndex);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AccessRestriction::DeleteEntry(size_t index, const FabricIndex fabricIndex)
+{
+    ChipLogProgress(DataManagement, "AccessRestriction: delete entry f=%u i=%lu", fabricIndex, index);
+
+    auto entry = GetEntry(fabricIndex, index);
+    if (entry == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+    else
+    {
+        mFabricEntries[fabricIndex].erase(mFabricEntries[fabricIndex].begin() + static_cast<int>(index));
+
+        for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+        {
+            listener->OnEntryChanged(fabricIndex, index, entry, EntryListener::ChangeType::kRemoved);
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AccessRestriction::UpdateEntry(size_t index, const Entry & entry, const FabricIndex fabricIndex)
+{
+    ChipLogProgress(DataManagement, "AccessRestriction: update entry f=%u i=%lu", fabricIndex, index);
+
+    auto localEntry = GetEntry(fabricIndex, index);
+    if (localEntry != nullptr)
+    {
+        *localEntry = entry;
+
+        for (EntryListener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+        {
+            listener->OnEntryChanged(fabricIndex, index, localEntry, EntryListener::ChangeType::kUpdated);
+        }
+
+        return CHIP_NO_ERROR;
+    }
+
+    return CHIP_ERROR_NOT_FOUND;
+}
+
+CHIP_ERROR AccessRestriction::Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath)
+{
+    ChipLogProgress(DataManagement, "AccessRestriction: action %d", static_cast<int>(requestPath.requestType));
+
+    if (requestPath.requestType == RequestType::kRequestTypeUnknown)
+    {
+        ChipLogError(DataManagement, "AccessRestriction: RequestPath type is unknown");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // wildcard event subscriptions are allowed since wildcard is only used when setting up the subscription and
+    // we want that request to succeed (when generating the report, this method will be called with the specific
+    // event id). All other requests require an entity id
+    if (!requestPath.entityId.has_value())
+    {
+        if (requestPath.requestType == RequestType::kEventReadOrSubscribeRequest)
+        {
+            return CHIP_NO_ERROR;
+        }
+        else
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+    }
+
+    for (auto & entry : mFabricEntries[subjectDescriptor.fabricIndex])
```

Comment:
This does not work for commissioning - you will need to check specifically for the pase fabric index, but then ALSO check whether there is an ongoing commissioning on this fabric index in order to defer to the CommissioningARL entry.

Reply:
This should be fixed now.